lib/node/runner/checker.go

```diff
@@ -628,30 +639,66 @@ func saveBlock(checker *BallotChecker) error {
 	return nil
 }
 
-func reorganizeTransactionPool(checker *BallotChecker, basisIndex string, proposer string) error {
+func reorganizeTransactionPool(checker *BallotChecker) error {
+	basisIndex := checker.Ballot.VotingBasis().Index()
+	proposer := checker.Ballot.Proposer()
+	transactionPool := checker.NodeRunner.TransactionPool
 	rr, found := checker.NodeRunner.Consensus().RunningRounds[basisIndex]
 	if found {
-		checker.NodeRunner.TransactionPool.Remove(rr.Transactions[proposer]...)
+		transactionPool.Remove(rr.Transactions[proposer]...)
 	}
 
-	// [TODO] TxValidate() again
+	invalids := []string{}
+	for source := range checker.LatestUpdatedSources {
+		tx, found := transactionPool.GetFromSource(source)
```

Comment:
@spikeekips Do we need transactions cache here?

Reply:
@Charleslee522 As you know, `TransactionCache` handles `transaction.Pool` and `block.TransactionPool`. This is only for `transaction.Pool`, `TransactionCache` is not good for here.